pkg/iam.go

```diff
@@ -102,13 +94,29 @@ func (e *IAMExporter) CollectLoop() {
 }
 
 // getIAMRoleCount returns number of IAM roles using IAMClient
-func getIAMRoleCount(ctx context.Context, client IAMClient) (int, error) {
+func getIAMRoleCount(ctx context.Context, client awsclient.Client) (int, error) {
 	var count int
-	err := client.ListRolesPagesWithContext(ctx, &iam.ListRolesInput{
-		MaxItems: aws.Int64(1000),
-	}, func(output *iam.ListRolesOutput, _ bool) bool {
+	var marker *string
+	
+	for {
+		input := &iam.ListRolesInput{
+			MaxItems: aws.Int32(1000),
+		}
+		if marker != nil {
+			input.Marker = marker
+		}
+
+		output, err := client.ListRoles(ctx, input)
+		if err != nil {
+			return 0, err
+		}
+		
 		count += len(output.Roles)
-		return true
-	})
-	return count, err
+		
+		if !output.IsTruncated {
+			break
+		}
+		marker = output.Marker
+	}
+	return count, nil
```

Comment:
should use [NewListRolesPaginator](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam#NewListRolesPaginator) in `client` to handle pagination, please also check all other list methods are using paginator.

Reply:
yes, it is not necessary to grab all the details. Fixed in 6ff09a2891fef30f8b3138b98fa9690c9e3ed893

Reply:
should use [NewListRolesPaginator](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam#NewListRolesPaginator), I'll write a correct version tomorrow

Reply:
`go generate` didn't regenerate all files somehow, so I manually run `mockgen -source=./awsclient.go -destination=./mock/zz_generated.mock_client.go -package=moc` in pkg/awsclient, this caused a diff https://github.com/app-sre/aws-resource-exporter/pull/191/files#diff-3a4a23fa81b01b3a138a03668998b271d37566be70b438f35a026818297c2da0L2, I don't know if it matters

Reply:
I run that command (using `-package=mock`, it looks to be a typo there) or the more generic `go generate ./pkg/awsclient/` and I get no diff 🤷